src/librustc_mir/def_use.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
```

Comment:
I think this is more of a "mention analysis" than a proper def-use chain, which is more of a concept with SSA.